crates/ra_assists/src/handlers/introduce_variable.rs

```diff
@@ -65,28 +64,43 @@ pub(crate) fn introduce_variable(acc: &mut Assists, ctx: &AssistContext) -> Opti
             if full_stmt.unwrap().semicolon_token().is_none() {
                 buf.push_str(";");
             }
-            edit.replace(expr.syntax().text_range(), buf);
-        } else {
-            buf.push_str(";");
-
-            // We want to maintain the indent level,
-            // but we do not want to duplicate possible
-            // extra newlines in the indent block
-            let text = indent.text();
-            if text.starts_with('\n') {
-                buf.push_str("\n");
-                buf.push_str(text.trim_start_matches('\n'));
-            } else {
-                buf.push_str(text);
+            let offset = expr.syntax().text_range();
+            match ctx.config.snippet_cap {
+                Some(cap) => {
+                    let snip = buf.replace("let var_name", "let $0var_name");
```

Comment:
With snipped support shouldn't this be `let ${1:var_name}` so that `var_name` can be set immediately in the UI or am I missing something?

Reply:
Yeah, I've thought about this. We do want to use linked edit capability of snippets. The problem is that we implement snippets on the client side manually, to avoid fighting over indentation, and adding linked edits support there would be a lot of work. So, I'd just either:

* punt on this until upstream support
* check how typescript handles this